homeassistant/components/srp_energy/sensor.py

```diff
@@ -93,28 +102,49 @@ async def async_update_data():
     # Fetch initial data so we have data when entities subscribe
     await coordinator.async_refresh()
 
-    async_add_entities([SrpEntity(coordinator)])
+    async_add_entities(
+        [
+            SrpEntity(
+                coordinator,
+                config_name=config_name,
+            )
+        ]
+    )
 
 
 class SrpEntity(SensorEntity):
     """Implementation of a Srp Energy Usage sensor."""
 
+    _attr_has_entity_name = True
     _attr_attribution = "Powered by SRP Energy"
     _attr_icon = "mdi:flash"
     _attr_should_poll = False
 
-    def __init__(self, coordinator) -> None:
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        config_name: str,
+    ) -> None:
         """Initialize the SrpEntity class."""
         self._name = SENSOR_NAME
         self.type = SENSOR_TYPE
         self.coordinator = coordinator
         self._unit_of_measurement = UnitOfEnergy.KILO_WATT_HOUR
         self._state = None
-
-    @property
-    def name(self) -> str:
-        """Return the name of the sensor."""
-        return f"{DEFAULT_NAME} {self._name}"
+        self._config_name = config_name
+        unique_id: str = f"{config_name}_energy_usage".lower()
+        LOGGER.debug("Setting unique id %s", unique_id)
+        self._attr_unique_id = unique_id
+
+        # Configure device
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, f"{config_name}_Energy usage")},
+            configuration_url=DEVICE_CONFIG_URL,
+            manufacturer=DEVICE_MANUFACTURER,
+            model=DEVICE_MODEL,
+            name=f"{config_name} Energy consumption",
+        )
```

Comment:
Should be removed as not related to this PR

Reply:
'm following the same pattern as Meteorologisk institutt (Met.no) when configuring two different entries. If a name isn't provided, the Location is used as the default. This goes into creating the unique name of the sensor which requires the service/device information as well to separate the different sensors.

Reply:
No, you don't need to create a device to set name or unique id.